---
title: "qpCR-Figures"
author: "C-T Berezin"
date: "10/30/2021"
output:
  pdf_document: default
  html_document: default
---

```{r}
library(tidyverse)
library(ggplot2)
library(ggthemes)
library(forcats)
library(ggpubr)
library(writexl)
```

### Is POMC under circadian regulation? 

```{r pomc}
pomc <- read.csv("../data/08202021-POMC-final.csv", fileEncoding = 'UTF-8-BOM')
pomc <- pomc %>% mutate(Time = as.factor(Time)) %>% 
  mutate(Time = fct_relevel(Time, c("8am", "4pm", "12am"))) %>%
  mutate(Time = fct_recode(Time,
                           "08:00" = "8am",
                           "16:00" = "4pm",
                           "00:00" = "12am")) %>% 
  filter(Relative.GE < 3) #remove mouse with high relative GE (data not normal)

#p-values from SigmaPlot
pomc_stats <- tibble::tribble(
  ~group1, ~group2, ~group3, ~p.adj,
  "08:00", "16:00", "00:00", "0.865"
)


pomc_plot <- ggplot(pomc, aes(x=Time, y=Relative.GE)) +
  geom_point() +
  stat_summary(fun = 'mean', geom="bar", aes(alpha=0.5), fill="darkorange") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width=.1, alpha=0.5) +
  theme_few() +
  theme(legend.position = "none") +
  labs(x="Time (Hour)",
       y="Relative Gene Expression",
       title = "Figure 3",
       subtitle = "Expression of POMC mRNA") +
  geom_bracket(xmin="08:00", xmax="00:00", label=pomc_stats$p.adj, data=pomc_stats, y.position=2.5) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
  

ggsave(filename="../figures/pomc_fig_w_bars.png", plot=pomc_plot, height=3, width=3)
```


### McKO validation

```{r mcko}
mcko <- read.csv("../data/08162021-McKO-totalret-final.csv", fileEncoding = 'UTF-8-BOM')
mcko <- mcko %>% 
  mutate(Genotype = fct_relevel(Genotype, c("WT", "McKO")))

#p-values from SigmaPlot
mcko_stats <- tibble::tribble(
  ~group1, ~group2, ~p.adj,
  "WT", "McKO", "<0.001"
)

mcko_plot <- mcko %>% 
  ggplot(aes(x=Genotype, y=Relative.GE)) +
  geom_point() +
  stat_summary(fun = 'mean', geom="bar", aes(alpha=0.5), fill="darkorange") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width=.1, alpha=0.5) +
  theme_few() +
  theme(legend.position = "none") +
  labs(y="Relative Gene Expression",
       title="Expression of MOR mRNA") +
  stat_pvalue_manual(mcko_stats,
    y.position = 1.5, step.increase = 0.1,
    label = "p.adj") +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
ggsave(filename="../figures/mcko_fig_bars_orange.png", plot=mcko_plot, height=4, width=3)
```

### MOR mRNA in morphine vs saline treatment (retinas only)

```{r mor morphine}
treatment <- c("saline", "saline", "saline", "saline",
               "morphine", "morphine", "morphine", "morphine")
sample_num <- c("1","2","3","4",
                "1","2","3","4")
rge <- c(1.025,
         1.054,
         1.12,
         0.822,
         0.677,
         1.054,
         0.938,
         1.057)
mor <- tibble(treatment, sample_num, rge)
mor <- mor %>% 
  mutate(treatment = fct_relevel(treatment, c("saline", "morphine")))

#p-value from SigmaPlot
mor_stats <- tibble::tribble(
  ~group1, ~group2, ~p.adj,
  "saline", "morphine", "0.528"
)

mor_plot <- mor %>% ggplot(aes(x=treatment, y=rge)) +
  stat_summary(fun = 'mean', geom="bar", fill="grey40") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width=.1, alpha=0.5) +
  geom_point() +
  labs(title ="Expression of MOR mRNA",
       x="",
       y="Relative Gene Expression") +
  theme_few() +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +
  stat_pvalue_manual(mor_stats,
    y.position = 1.25, step.increase = 0.1,
    label = "p.adj")


ggsave(filename="../figures/mor_morphine_qpcr.png", plot=mor_plot, height=4, width=3)
```
