---
title: "qpCR-Figures"
author: "C-T Berezin"
date: "10/30/2021"
output:
  pdf_document: default
  html_document: default
---

```{r, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, error = FALSE)
library(tidyverse)
library(ggplot2)
library(ggthemes)
library(forcats)
library(ggpubr)
library(writexl)
```

### Is POMC under circadian regulation? 

```{r pomc-circadian}
pomc_circ <- read.csv("../data/08202021-POMC-circ-final.csv",
                      fileEncoding = 'UTF-8-BOM')
pomc_circ <- pomc_circ %>% mutate(Time = as.factor(Time),
                                  Sex = as.factor(Sex)) %>% 
  mutate(Time = fct_relevel(Time, c("8am", "4pm", "12am"))) %>%
  mutate(Time = fct_recode(Time,
                           "2" = "8am",
                           "10" = "4pm",
                           "18" = "12am")) %>% 
  filter(Relative.GE < 3) #remove mouse with high relative GE (data not normal)

#p-values from SigmaPlot
pomc_circ_stats <- tibble::tribble(
  ~group1, ~group2, ~group3, ~p.adj,
  "08:00", "16:00", "00:00", "0.865"
)


pomc_circ_plot <- ggplot(pomc_circ, aes(x=Time, y=Relative.GE)) +
  stat_summary(fun = 'mean', geom="bar") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width=.2) +
  geom_point(aes(col=Sex)) +
  theme_few() +
  labs(x="Zeitgeber Time (ZT)",
       y="Relative Gene Expression") +
  geom_bracket(xmin="2", xmax="18", y.position=2.5, label="n.s.") +
  geom_bracket(xmin="10", xmax="18", y.position=2.5, label="") +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
  
pomc_circ_plot

#ggsave(filename="../figures/pomc_circ_fm.png", plot=pomc_circ_plot, height=3, width=3)
```

```{r}
#males only
pomc_circ_males <- pomc_circ %>% filter(Sex == "M")

pomc_circ_m_plot <- ggplot(pomc_circ_males, aes(x=Time, y=Relative.GE)) +
  stat_summary(fun = 'mean', geom="bar") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width=.2) +
  geom_point() +
  theme_few() +
  labs(x="Zeitgeber Time (ZT)",
       y="Relative Gene Expression") +
  geom_bracket(xmin="2", xmax="18", y.position=2.2, label="n.s.") +
  geom_bracket(xmin="10", xmax="18", y.position=2.2, label="") +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

pomc_circ_m_plot

#ggsave(filename="../figures/pomc_circ_m.png", plot=pomc_circ_m_plot, height=3, width=3)

#normal
shapiro.test(pomc_circ_males$Relative.GE)

#non-homogenous variance
car::leveneTest(Relative.GE ~ Time, pomc_circ_males)

pomc_circ_m_lm <- lm(Relative.GE ~ Time, pomc_circ_males)

#res vs fitted plots looks OK, non-homogenous variance probably not a big concern
plot(pomc_circ_m_lm, which=c(1,2))

anova(pomc_circ_m_lm)
#no significant comps
emmeans::emmeans(pomc_circ_m_lm, pairwise ~ Time)$contrasts

ctrl_pomc_circ_males <- pomc_circ_males %>% filter(Time == "2")

pomc_circ_m_sumstats <- pomc_circ_males %>%
  group_by(Time) %>%
  summarise(n = n(),
            mean = mean(Relative.GE))

pomc_circ_m_sumstats

#between var
sum(5*(mean(pomc_circ_males$Relative.GE - mean(ctrl_pomc_circ_males$Relative.GE)))^2)
#within.var
anova(pomc_circ_m_lm)["Residuals", "Mean Sq"]
power.anova.test(groups=3, n=5, between.var=0.009116243,
                 within.var=0.1952332, sig.level=0.05)
```

### Is POMC expression light-driven?

```{r pomc-light-dark}
pomc_ld <- read.csv("../data/01142022-POMC-light-dark-add'l-samps.csv", fileEncoding = 'UTF-8-BOM')
pomc_ld <- pomc_ld %>% mutate(Time = as.factor(Time.of.Day),
                              Light = as.factor(Light.Condition),
                              Sex = as.factor(Sex)) %>% 
                      mutate(Light = fct_relevel(Light, c("light", "dark"))) %>% 
                      dplyr::select(-Time.of.Day, -Light.Condition)


pomc_ld_mf <- ggplot(pomc_ld, aes(x=Light, y=Relative.G.E.)) +
  theme_few() +
  facet_wrap(~Time) +
  stat_summary(fun = 'mean', geom="bar") +
  geom_point(aes(col=Sex)) +
  labs(x="",
       y="Relative Gene Expression",
       title = "POMC mRNA Expression")
  
pomc_ld_mf

#ggsave(filename="../figures/pomc_ld_mf.png", plot=pomc_ld_mf, height=3, width=3)

shapiro.test(pomc_ld$Relative.G.E.)

#non-homogenous variance!
car::leveneTest(Relative.G.E. ~ Sex*Light*Time, pomc_ld)

#doesn't meet assumptions, take care
pomc_ld_lm <- lm(Relative.G.E. ~ Sex*Light*Time, pomc_ld)

qqnorm(pomc_ld_lm$residuals)
qqline(pomc_ld_lm$residuals)

anova(pomc_ld_lm)
#sex is significant factor
pomc_ld_av <- aov(pomc_ld_lm)
TukeyHSD(pomc_ld_av)
```

```{r}
#females only
pomc_ld_f <- pomc_ld %>% filter(Sex == "F")

pomc_ld_f_plot <- ggplot(pomc_ld_f, aes(x=Light, y=Relative.G.E.)) +
  theme_few() +
  facet_wrap(~Time) +
  stat_summary(fun = 'mean', geom="bar") +
  geom_point() +
  labs(x="",
       y="Relative Gene Expression",
       title = "POMC mRNA in Females") +
  scale_y_continuous(limits = c(0,3))

pomc_ld_f_plot

#ggsave(filename="../figures/pomc_ld_f.png", plot=pomc_ld_f_plot, height=3, width=3)
```

```{r}
#males only
pomc_ld_m <- pomc_ld %>% filter(Sex == "M")

pomc_ld_m_plot <- ggplot(pomc_ld_m, aes(x=Light, y=Relative.G.E.)) +
  theme_few() +
  facet_wrap(~Time) +
  stat_summary(fun = 'mean', geom="bar", aes(fill=Light)) +
  stat_summary(fun.data = mean_se, geom = "errorbar", width=.2) +
  geom_point() +
  labs(x="",
       y="Relative Gene Expression") +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +
  scale_fill_manual(values=c("yellow", "grey70"))

pomc_ld_m_plot

#ggsave(filename="../figures/pomc_ld_m.png", plot=pomc_ld_m_plot, height=3, width=3)

#non-normal
shapiro.test(pomc_ld_m$Relative.G.E.)
#normal
shapiro.test(log(pomc_ld_m$Relative.G.E.))

#non-homogenous variance, take care
car::leveneTest(log(Relative.G.E.) ~ Light*Time, pomc_ld_m)

pomc_ld_m_log_lm <- lm(log(Relative.G.E.) ~ Light*Time, pomc_ld_m)

#res vs fitted plots looks OK, non-homogenous variance probably not a big concern
plot(pomc_ld_m_log_lm, which=c(1,2))

anova(pomc_ld_m_log_lm)
#time is a significant factor
#but no significant pairwise comps
emmeans::emmeans(pomc_ld_m_log_lm, pairwise ~ Light * Time)$contrasts
emmeans::emmeans(pomc_ld_m_log_lm, pairwise ~ Light | Time)$contrasts
emmeans::emmeans(pomc_ld_m_log_lm, pairwise ~ Time | Light)$contrasts

pomc_ld_m_log_plot <- ggplot(pomc_ld_m, aes(x=Light, y=log(Relative.G.E.))) +
  theme_few() +
  facet_wrap(~Time) +
  stat_summary(fun = 'mean', geom="bar", aes(fill=Light)) +
  stat_summary(fun.data = mean_se, geom = "errorbar", width=.2) +
  geom_point() +
  labs(x="",
       y="Relative Gene Expression") +
  ylim(ymin=-0.75, ymax=1.5) +
  scale_fill_manual(values=c("yellow", "grey70"))
pomc_ld_m_log_plot

#ggsave(filename="../figures/pomc_ld_m_log.png", plot=pomc_ld_m_log_plot, height=3, width=3)

ctrl_pomc_ld_m <- pomc_ld_m %>% filter(Light == "light" & Time == "day")

pomc_ld_m_sumstats <- pomc_ld_m %>%
  group_by(Light, Time) %>%
  summarise(n = n(),
            mean = mean(Relative.G.E.),
            log_mean = mean(log(Relative.G.E.)))

pomc_ld_m_sumstats

#between var
sum(3*(mean(pomc_ld_m$Relative.G.E. - mean(ctrl_pomc_ld_m$Relative.G.E.)))^2)
#within.var
anova(pomc_ld_m_log_lm)["Residuals", "Mean Sq"]
power.anova.test(groups=4, n=3, between.var=1.411641,
                 within.var=0.3873501, sig.level=0.05)
```


### POMC circadian + light-dark

```{r}
pomc_all <- read.csv("../data/02032022-POMC-circ-ld-gene-study.csv",
                     fileEncoding = 'UTF-8-BOM')
pomc_all <- pomc_all %>% mutate(Sex = as.factor(Sex),
                                Hour = as.factor(Hour),
                                Time = as.factor(Time),
                                Light = as.factor(Light))
pomc_all <- pomc_all %>% mutate(Hour = fct_relevel(Hour, c("8am", "12pm", "4pm", "12am", "1am")),
                                Light = fct_relevel(Light, c("light", "dark")))

qqnorm(pomc_all$RelativeGE)
qqline(pomc_all$RelativeGE)

qqnorm(log2(pomc_all$RelativeGE))
qqline(log2(pomc_all$RelativeGE))
shapiro.test(log2(pomc_all$RelativeGE))

pomc_all_lm <- lm(log2(RelativeGE) ~ Sex * Light * Time, data=pomc_all)
plot(pomc_all_lm, which=c(1))
pomc_all_aov <- anova(pomc_all_lm)
pomc_all_aov
#TukeyHSD(aov(pomc_all_lm))

#but can we actually group all timepoints within day and night
day_pomc_all  <- pomc_all %>% filter(Time == "day")
qqnorm(day_pomc_all$RelativeGE)
qqline(day_pomc_all$RelativeGE)

qqnorm(log2(day_pomc_all$RelativeGE))
qqline(log2(day_pomc_all$RelativeGE))
shapiro.test(log2(day_pomc_all$RelativeGE))
day_pomc_lm <- lm(log2(RelativeGE) ~ Hour, data=day_pomc_all)
anova(day_pomc_lm)

night_pomc_all  <- pomc_all %>% filter(Time == "night")
qqnorm(night_pomc_all$RelativeGE)
qqline(night_pomc_all$RelativeGE)

qqnorm(log2(night_pomc_all$RelativeGE))
qqline(log2(night_pomc_all$RelativeGE))
shapiro.test(log2(night_pomc_all$RelativeGE))
night_pomc_lm <- lm(log2(RelativeGE) ~ Hour, data=night_pomc_all)
anova(night_pomc_lm)
t.test(RelativeGE ~ Hour, data=night_pomc_all)
#indeed

par(mfrow=c(1,2))

pomc_all_mf <- ggplot(pomc_all, aes(x=Time, y=RelativeGE)) +
  theme_few() +
  facet_wrap(~Sex) +
  stat_summary(fun = 'mean', geom="bar") +
  geom_point(aes(col=Light))

pomc_all_mf

#ggsave(filename="../figures/pomc_all_mf.png", plot=pomc_all_mf)
```

```{r}
#males only
m_pomc_all <- pomc_all %>% filter(Sex == "M")

m_pomc_all %>% group_by(Time, Light) %>% 
  summarise(n = n(),
            mean = mean(RelativeGE))

pomc_all_m <- ggplot(m_pomc_all, aes(x=Light, y=RelativeGE)) +
  stat_summary(fun = 'mean', geom="bar", aes(fill=Light)) +
  geom_point(size=2) +
  facet_wrap(~Time) +
  stat_summary(fun.data = mean_se, geom = "errorbar", width=.1) +
  theme_bw() %+replace%
  theme(legend.position="none") +
  scale_fill_manual(values=c("yellow", "grey70")) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) #+
#  geom_signif(y_position=4.8, xmin=1, xmax=2, annotation="n.s.", tip_length=0.02)

pomc_all_m

#ggsave(filename="../figures/pomc_all_m.png", plot=pomc_all_m, height=4, width=6)

#non-normal
shapiro.test(m_pomc_all$RelativeGE)
#normal
shapiro.test(log(m_pomc_all$RelativeGE))

car::leveneTest(log(RelativeGE) ~ Light*Time, m_pomc_all)

pomc_all_m_log_lm <- lm(log(RelativeGE) ~ Light*Time, m_pomc_all)

plot(pomc_all_m_log_lm, which=c(1,2))

#no significant factors
anova(pomc_all_m_log_lm)

emmeans::emmeans(pomc_all_m_log_lm, pairwise ~ Light * Time)$contrasts
emmeans::emmeans(pomc_all_m_log_lm, pairwise ~ Light | Time)$contrasts
emmeans::emmeans(pomc_all_m_log_lm, pairwise ~ Time | Light)$contrasts

pomc_all_m_log <- ggplot(m_pomc_all, aes(x=Light, y=log(RelativeGE))) +
  stat_summary(fun = 'mean', geom="bar", aes(fill=Light)) +
  geom_point(size=2) +
  facet_wrap(~Time) +
  stat_summary(fun.data = mean_se, geom = "errorbar", width=.1) +
  theme_bw() %+replace%
  theme(legend.position="none") +
  scale_fill_manual(values=c("yellow", "grey70")) #+
#  geom_signif(y_position=4.8, xmin=1, xmax=2, annotation="n.s.", tip_length=0.02)

pomc_all_m_log

#ggsave(filename="../figures/pomc_all_m_log.png", plot=pomc_all_m_log, height=4, width=3)

ctrl_pomc_all_m <- m_pomc_all %>% filter(Light == "light" & Time == "day")

pomc_all_m_sumstats <- m_pomc_all %>%
  group_by(Light, Time) %>%
  summarise(n = n(),
            mean = mean(RelativeGE),
            log_mean = mean(log(RelativeGE)))

pomc_all_m_sumstats

#between var
sum(3*(mean(m_pomc_all$RelativeGE - mean(ctrl_pomc_all_m$RelativeGE)))^2)
#within.var
anova(pomc_all_m_log_lm)["Residuals", "Mean Sq"]
power.anova.test(groups=4, n=3, between.var=0.0006157598,
                 within.var=0.2920105, sig.level=0.05)
```

### McKO validation

```{r mcko}
mcko <- read.csv("../data/08162021-McKO-totalret-final.csv", fileEncoding = 'UTF-8-BOM')
mcko <- mcko %>% 
  mutate(Genotype = fct_relevel(Genotype, c("WT", "McKO")))

#p-values from SigmaPlot
mcko_stats <- tibble::tribble(
  ~group1, ~group2, ~p.adj,
  "WT", "McKO", "<0.001"
)

mcko_plot <- mcko %>% 
  ggplot(aes(x=Genotype, y=Relative.GE, fill=Genotype)) +
  stat_summary(fun = 'mean', geom="bar") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width=.1) +
  geom_point() +
  theme_bw() %+replace%
  theme(legend.position = "none") +
  labs(y="Relative Gene Expression") +
  geom_signif(y_position=1.55, xmin=1, xmax=2, annotation="***", tip_length=0.02) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

mcko_plot

#ggsave(filename="../figures/mcko_fig_bars.png", plot=mcko_plot, height=4, width=3)

mcko %>% group_by(Genotype) %>% 
  summarise(mean(Relative.GE))
```

### MOR mRNA in morphine vs saline treatment (retinas only)

```{r mor morphine}
treatment <- c("saline", "saline", "saline", "saline",
               "morphine", "morphine", "morphine", "morphine")
sample_num <- c("1","2","3","4",
                "1","2","3","4")
rge <- c(1.025,
         1.054,
         1.12,
         0.822,
         0.677,
         1.054,
         0.938,
         1.057)
mor <- tibble(treatment, sample_num, rge)
mor <- mor %>% 
  mutate(treatment = fct_relevel(treatment, c("saline", "morphine")))

#p-value from SigmaPlot
mor_stats <- tibble::tribble(
  ~group1, ~group2, ~p.adj,
  "saline", "morphine", "0.528"
)

mor_plot <- mor %>% ggplot(aes(x=treatment, y=rge)) +
  stat_summary(fun = 'mean', geom="bar", fill="grey40") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width=.1, alpha=0.5) +
  geom_point() +
  labs(title ="Expression of MOR mRNA",
       x="",
       y="Relative Gene Expression") +
  theme_few() +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +
  stat_pvalue_manual(mor_stats,
    y.position = 1.25, step.increase = 0.1,
    label = "p.adj")

mor_plot

#ggsave(filename="../figures/mor_morphine_qpcr.png", plot=mor_plot, height=4, width=3)
```