---
title: "qpCR-Figures"
author: "C-T Berezin"
date: "10/30/2021"
output:
  pdf_document: default
  html_document: default
---

```{r, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, error = FALSE)
library(tidyverse)
library(ggplot2)
library(ggthemes)
library(forcats)
library(ggpubr)
library(writexl)
library(svglite)
library(lme4)
library(lmerTest)
library(emmeans)
```

### McKO validation

```{r mcko}
mcko <- read.csv("../data/08162021-McKO-totalret-final.csv", fileEncoding = 'UTF-8-BOM')
mcko <- mcko %>% 
  mutate(Genotype = fct_relevel(Genotype, c("WT", "McKO")))

mcko_summ <- mcko %>% group_by(Genotype) %>%
                      summarise(n = n(),
                                mean = mean(Relative.GE),
                                sd = sd(Relative.GE))

mcko_plot <- ggplot() +
  stat_summary(data=mcko, aes(x=Genotype, y=Relative.GE, fill=Genotype),
               fun = 'mean', geom="bar", width=0.75) +
  geom_errorbar(data=mcko_summ, aes(x=Genotype, ymin=mean-sd, ymax=mean+sd), width=0.2) +
  geom_point(data=mcko, aes(x=Genotype, y=Relative.GE), size=2) +
  theme_bw() %+replace%
  theme(legend.position = "none") +
  labs(y="Relative Gene Expression",
       x="") +
  scale_y_continuous(expand = expansion(mult = c(0, 0.2)))

mcko_plot

#ggsave(filename="../figures/mcko_fig_bars.svg", plot=mcko_plot, height=4, width=3)

mcko %>% group_by(Genotype) %>% 
  summarise(n= n(),
            mean = mean(Relative.GE),
            sd = sd(Relative.GE))

t.test(mcko$Relative.GE ~ mcko$Genotype)
```

### McKO validation, retina & hypothalamus

```{r}
mor <- read.csv("../data/MOR-hypothalamus-results.csv", fileEncoding = 'UTF-8-BOM')
mor <- mor %>% 
  mutate(Genotype = fct_relevel(Genotype, c("WT", "McKO")),
         Tissue = fct_relevel(Tissue, c("retina", "hypothalamus"))) %>% 
  filter(Sample != "WT5-R" & Sample != "WT5-H")

mor_summ <- mor %>% group_by(Genotype, Tissue) %>%
                      summarise(n = n(),
                                mean = mean(RelativeGE),
                                sd = sd(RelativeGE))

mor_plot <- ggplot() +
  stat_summary(data=mor, aes(x=Genotype, y=RelativeGE, fill=Genotype),
               fun = 'mean', geom="bar", width=0.75) +
  geom_errorbar(data=mor_summ, aes(x=Genotype, ymin=mean-sd, ymax=mean+sd), width=0.2) +
  geom_point(data=mor, aes(x=Genotype, y=RelativeGE), size=2) +
  facet_wrap(~Tissue, scales="free") +
  theme_bw() %+replace%
  theme(legend.position = "none") +
  labs(y="Relative Gene Expression",
       x="") +
  scale_y_continuous(expand = expansion(mult = c(0, 0.25)))

mor_plot

#ggsave(filename="../figures/mor_retina_hypothalamus.svg", plot=mor_plot, height=4, width=6)
```

```{r}
mor <- mor %>% mutate(Sample = str_sub(Sample, start=1L, end=-3L))
#bad
mor_lmer <- lmer(RelativeGE ~ Genotype * Tissue + (1 | Sample), mor)
plot(mor_lmer, type=c("p","smooth"), col.line=1)
lattice::qqmath(mor_lmer)
#much better
mor_log_lmer <- lmer(log(RelativeGE) ~ Genotype * Tissue + (1 | Sample), mor)
plot(mor_log_lmer, type=c("p","smooth"), col.line=1)
lattice::qqmath(mor_log_lmer)

anova(mor_log_lmer)
emmeans(mor_log_lmer, pairwise ~ Genotype | Tissue)$contrasts
emmeans(mor_log_lmer, pairwise ~ Tissue | Genotype)$contrasts


mor <- mor %>% mutate(log_rge = log(RelativeGE))

mor_summ$log_mean <- mor %>% group_by(Genotype, Tissue) %>%
  summarise(log_mean = mean(log_rge)) %>%
  ungroup() %>%
  pull(log_mean)

mor_summ$log_sd <- mor %>% group_by(Genotype, Tissue) %>%
  summarise(log_sd = sd(log_rge)) %>%
  ungroup() %>%
  pull(log_sd)


# log_sd = sd(log_rge)

mor_log_plot <- ggplot() +
  stat_summary(data=mor, aes(x=Genotype, y=log_rge, fill=Genotype),
               fun = 'mean', geom="bar", width=0.75) +
  geom_errorbar(data=mor_summ, aes(x=Genotype, ymin=log_mean-log_sd, ymax=log_mean+log_sd), width=0.2) +
  geom_point(data=mor, aes(x=Genotype, y=log_rge), size=2) +
  facet_wrap(~Tissue) +
  theme_bw() %+replace%
  theme(legend.position = "none") +
  labs(y="Relative Gene Expression",
       x="") +
  ylim(-1, 3.3)

mor_log_plot

#ggsave(filename="../figures/mor_log.svg", plot=mor_log_plot, height=4, width=6)
```


### Is POMC under circadian regulation? 

```{r pomc-circadian}
pomc_circ <- read.csv("../data/08202021-POMC-circ-final.csv",
                      fileEncoding = 'UTF-8-BOM')
pomc_circ <- pomc_circ %>% mutate(Time = as.factor(Time),
                                  Sex = as.factor(Sex)) %>% 
  mutate(Time = fct_relevel(Time, c("8am", "4pm", "12am"))) %>%
  mutate(Time = fct_recode(Time,
                           "2" = "8am",
                           "10" = "4pm",
                           "18" = "12am")) %>% 
  filter(Relative.GE < 3) #remove mouse with high relative GE (data not normal)

pomc_circ_plot <- ggplot(pomc_circ, aes(x=Time, y=Relative.GE)) +
  stat_summary(fun = 'mean', geom="bar") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width=.2) +
  geom_point(aes(col=Sex)) +
  theme_few() +
  labs(x="Zeitgeber Time (ZT)",
       y="Relative Gene Expression") +
  geom_bracket(xmin="2", xmax="18", y.position=2.5, label="n.s.") +
  geom_bracket(xmin="10", xmax="18", y.position=2.5, label="") +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
  
pomc_circ_plot

#ggsave(filename="../figures/pomc_circ_fm.png", plot=pomc_circ_plot, height=3, width=3)
```

```{r}
#males only
pomc_circ_males <- pomc_circ %>% filter(Sex == "M")

pomc_circ_m_summ <- pomc_circ_males %>% group_by(Time) %>%
                    summarise(n = n(),
                              mean = mean(Relative.GE),
                              sd = sd(Relative.GE))

pomc_circ_m_summ

pomc_circ_m_plot <- ggplot() +
  stat_summary(data=pomc_circ_males, aes(x=Time, y=Relative.GE), fun = 'mean', geom="bar",
               fill=c("yellow", "yellow", "grey70"),
               width=0.75) +
  geom_errorbar(data=pomc_circ_m_summ, aes(x=Time, ymin=mean-sd, ymax=mean+sd), width=.2) +
  geom_point(data=pomc_circ_males, aes(x=Time, y=Relative.GE), size = 2) +
  theme_bw() %+replace%
  theme(plot.background = element_rect(fill = "transparent",colour = NA),
        panel.background = element_rect(fill = "transparent",colour = NA)) +
  labs(x="Zeitgeber Time (ZT)",
       y="Relative Gene Expression") +
  scale_y_continuous(expand = expansion(mult = c(0, 0.2)))

pomc_circ_m_plot

#ggsave(filename="../figures/pomc_circ_m.svg", plot=pomc_circ_m_plot, height=4, width=4)

#normal
shapiro.test(pomc_circ_males$Relative.GE)

#homogenous variance
car::leveneTest(Relative.GE ~ Time, pomc_circ_males)

pomc_circ_m_lm <- lm(Relative.GE ~ Time, pomc_circ_males)
plot(pomc_circ_m_lm, which=c(1,2))

anova(pomc_circ_m_lm)
#no significant comps
emmeans::emmeans(pomc_circ_m_lm, pairwise ~ Time)$contrasts

pomc_circ_m_sumstats <- pomc_circ_males %>%
  group_by(Time) %>%
  summarise(n = n(),
            mean = mean(Relative.GE),
            sd = sd(Relative.GE))

pomc_circ_m_sumstats

ctrl_pomc_circ_males <- pomc_circ_males %>% filter(Time == "2")
#between var
sum(5*(mean(pomc_circ_males$Relative.GE - mean(ctrl_pomc_circ_males$Relative.GE)))^2)
#within.var
anova(pomc_circ_m_lm)["Residuals", "Mean Sq"]
power.anova.test(groups=3, n=5, between.var=0.009116243,
                 within.var=0.1952332, sig.level=0.05)
```

### POMC circadian + light-dark analysis

```{r}
pomc_all <- read.csv("../data/02032022-POMC-circ-ld-gene-study.csv",
                     fileEncoding = 'UTF-8-BOM')
pomc_all <- pomc_all %>% mutate(Sex = as.factor(Sex),
                                Hour = as.factor(Hour),
                                Time = as.factor(Time),
                                Light = as.factor(Light))
pomc_all <- pomc_all %>% mutate(Hour = fct_relevel(Hour, c("8am", "12pm", "4pm", "12am", "1am")),
                                Light = fct_relevel(Light, c("light", "dark")))

```

```{r}
#males only
m_pomc_all <- pomc_all %>% filter(Sex == "M")

m_pomc_all <- m_pomc_all %>% mutate(ZT = fct_relevel(Hour, c("8am", "12pm", "4pm", "12am", "1am"))) %>%
  mutate(ZT = fct_recode(ZT,
                           "2" = "8am",
                           "6" = "12pm",
                           "10" = "4pm",
                           "18" = "12am",
                           "19" = "1am"))

head(m_pomc_all)

m_pomc_all_summ <- m_pomc_all %>% group_by(ZT, Light) %>% 
  summarise(n = n(),
            mean = mean(RelativeGE),
            sd = sd(RelativeGE),
            log_mean = mean(log(RelativeGE)),
            log_sd = sd(log(RelativeGE))
            )

m_pomc_all_summ

pomc_all_m <- ggplot() +
  stat_summary(data=m_pomc_all, aes(x=ZT, y=RelativeGE, fill=Light),
               fun = 'mean', geom="bar", width=0.75, position="dodge") +
  geom_point(data= m_pomc_all, aes(x=ZT, y=RelativeGE, fill=Light), size=2, position=position_dodge(0.75)) +
  geom_errorbar(data=m_pomc_all_summ, aes(x=ZT, ymin=mean-sd, ymax=mean+sd, fill=Light), width=.2, position=position_dodge(0.75)) +
  theme_bw() %+replace%
  theme(legend.position="none",
        axis.text.x = element_text(size = 12),
        strip.text.x = element_text(size=14),
        axis.title.x = element_blank(),
        panel.spacing = unit(0,"pt"),
        plot.background = element_rect(fill = "transparent",colour = NA),
        panel.background = element_rect(fill = "transparent",colour = NA)) +
  scale_fill_manual(values=c("yellow", "grey70")) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +
  labs(y="Relative Gene Expression")

pomc_all_m
```

```{r}
#can we group all timepoints within light/day and dark/night
m_pomc_day  <- m_pomc_all %>% filter(Time == "day" & Light == "light")

shapiro.test(m_pomc_day$RelativeGE)
car::leveneTest(RelativeGE ~ ZT, m_pomc_day)
day_pomc_lm <- lm(RelativeGE ~ ZT, data=m_pomc_day)
plot(day_pomc_lm, which=c(1,2))

shapiro.test(log(m_pomc_day$RelativeGE))
car::leveneTest(log(RelativeGE) ~ ZT, m_pomc_day)
day_pomc_log_lm <- lm(log(RelativeGE) ~ ZT, data=m_pomc_day)
plot(day_pomc_log_lm, which=c(1,2))

anova(day_pomc_log_lm)
emmeans::emmeans(day_pomc_log_lm, pairwise ~ ZT)$contrasts

m_pomc_night  <- m_pomc_all %>% filter(Time == "night" & Light == "dark")

shapiro.test(log(m_pomc_night$RelativeGE))
car::leveneTest(log(RelativeGE) ~ ZT, m_pomc_night)
night_pomc_log_lm <- lm(log(RelativeGE) ~ ZT, data=m_pomc_night)
plot(night_pomc_log_lm, which=c(1,2))

anova(night_pomc_log_lm)
emmeans::emmeans(night_pomc_log_lm, pairwise ~ ZT)$contrasts

#we can group light/day & dark/night samples

m_pomc_all <- m_pomc_all %>% mutate(ZT_group = fct_recode(ZT, "ZTday" = "2",
                                                    "ZTday" = "6",
                                                    "ZTday" = "10",
                                                    "ZTnight" = "18",
                                                    "ZTnight" = "19"))
```


```{r}
shapiro.test(m_pomc_all$RelativeGE)
car::leveneTest(RelativeGE ~ ZT_group, m_pomc_all)
car::leveneTest(RelativeGE ~ Light, m_pomc_all)
pomc_lm <- lm(RelativeGE ~ Light * ZT_group, data=m_pomc_all)
plot(pomc_lm, which=c(1,2))

m_pomc_all$Light

shapiro.test(log(m_pomc_all$RelativeGE))
car::leveneTest(log(RelativeGE) ~ ZT_group, m_pomc_all)
car::leveneTest(log(RelativeGE) ~ Light, m_pomc_all)
pomc_log_lm <- lm(log(RelativeGE) ~ Light * ZT_group, data=m_pomc_all)
plot(pomc_log_lm, which=c(1,2))

shapiro.test(sqrt(m_pomc_all$RelativeGE))
car::leveneTest(sqrt(RelativeGE) ~ ZT_group, m_pomc_all)
car::leveneTest(sqrt(RelativeGE) ~ Light, m_pomc_all)
pomc_sqrt_lm <- lm(sqrt(RelativeGE) ~ Light * ZT_group, data=m_pomc_all)
plot(pomc_sqrt_lm, which=c(1,2))

anova(pomc_log_lm)

pomc_log_zt_wilcox <- wilcox.test(log(RelativeGE) ~ ZT_group, data= m_pomc_all)
pomc_log_zt_wilcox
pomc_log_light_wilcox <- wilcox.test(log(RelativeGE) ~ Light, data= m_pomc_all)
pomc_log_light_wilcox
wake_Zstat <- qnorm(wake_day$p.value/2)
abs(wake_Zstat)/sqrt(9.5)
```

```{r}
zt_group_summ <- m_pomc_all %>% group_by(ZT_group, Light) %>% 
  summarise(n = n(),
            mean = mean(RelativeGE),
            sd = sd(RelativeGE),
            log_mean = mean(log(RelativeGE)),
            log_sd = sd(log(RelativeGE))
            )

zt_group_summ

pomc_plot <- ggplot() +
  stat_summary(data=m_pomc_all,
               aes(x=Light, y=RelativeGE, fill=Light),
               fun = 'mean', geom="bar",
               width=0.75, position="dodge") +
  geom_point(data= m_pomc_all,
             aes(x=Light, y=RelativeGE, fill=Light),
             size=2,
             position=position_dodge(0.75)) +
  geom_errorbar(data=zt_group_summ,
                aes(x=Light, ymin=mean-sd, ymax=mean+sd, fill=Light),
                width=.2,
                position=position_dodge(0.75)) +
  facet_wrap(~ZT_group) +
  theme_bw() %+replace%
  theme(legend.position="none",
        axis.text.x = element_text(size = 12),
        strip.text.x = element_text(size=14),
        axis.title.x = element_blank(),
        panel.spacing = unit(0,"pt"),
        plot.background = element_rect(fill = "transparent",colour = NA),
        panel.background = element_rect(fill = "transparent",colour = NA)) +
  scale_fill_manual(values=c("yellow", "grey70")) +
  labs(y="Relative Gene Expression") +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))

#ggsave(filename="../figures/pomc_all_m.svg", plot=pomc_all_m_log, height=4, width=6)

pomc_plot

pomc_log_plot <- ggplot() +
  stat_summary(data=m_pomc_all,
               aes(x=Light, y=log(RelativeGE), fill=Light),
               fun = 'mean', geom="bar",
               width=0.75, position="dodge") +
  geom_point(data= m_pomc_all,
             aes(x=Light, y=log(RelativeGE), fill=Light),
             size=2,
             position=position_dodge(0.75)) +
  geom_errorbar(data=zt_group_summ, 
              aes(x=Light, ymin=log_mean-log_sd, ymax=log_mean+log_sd, fill=Light),
              width=.2, position=position_dodge(0.75)) +
  facet_wrap(~ZT_group) +
  theme_bw() %+replace%
  theme(legend.position="none",
        axis.text.x = element_text(size = 12),
        strip.text.x = element_text(size=14),
        axis.title.x = element_blank(),
        panel.spacing = unit(0,"pt"),
        plot.background = element_rect(fill = "transparent",colour = NA),
        panel.background = element_rect(fill = "transparent",colour = NA)) +
  scale_fill_manual(values=c("yellow", "grey70")) +
  labs(y="log(Relative Gene Expression)") +
  ylim(ymin=-0.75, ymax=2.2)

pomc_log_plot

#ggsave(filename="../figures/pomc_all_m_log.svg", plot=pomc_all_m_log, height=4, width=6)
```

### POMC in hypothalamus/retina

```{r}
pomc_hyp <- read.csv("../data/06092022-POMC-hypothalamus.csv",
                     fileEncoding = 'UTF-8-BOM')
pomc_hyp <- pomc_hyp %>% mutate(Tissue = as.factor(Tissue),
                                Condition = as.factor(Condition),
                                Sample = as.factor(str_sub(Sample, start=1L, end=-3L)))
pomc_hyp

pomc_hyp_summ <- pomc_hyp %>% group_by(Tissue, Condition) %>%
                      summarise(n = n(),
                                mean = mean(RGE),
                                sd = sd(RGE))
pomc_hyp_summ

pomc_hyp_plot <- ggplot() +
  stat_summary(data=pomc_hyp, aes(x=Condition, y=RGE, fill=Condition),
               fun = 'mean', geom="bar", width=0.75) +
  geom_errorbar(data=pomc_hyp_summ, aes(x=Condition, ymin=mean-sd, ymax=mean+sd), width=0.2) +
  geom_point(data=pomc_hyp, aes(x=Condition, y=RGE), size=2) +
  facet_wrap(~Tissue, scales="free") +
  theme_bw() %+replace%
  theme(legend.position = "none") +
  labs(y="Relative Gene Expression",
       x="") +
  scale_y_continuous(expand = expansion(mult = c(0, 0.25)))

pomc_hyp_plot


#bad
pomc_hyp_lmer <- lmer(RGE ~ Condition * Tissue + (1 | Sample), pomc_hyp)
plot(pomc_hyp_lmer, type=c("p","smooth"), col.line=1)
lattice::qqmath(pomc_hyp_lmer)
#better
pomc_hyp_log_lmer <- lmer(log(RGE) ~ Condition * Tissue + (1 | Sample), pomc_hyp)
plot(pomc_hyp_log_lmer, type=c("p","smooth"), col.line=1)
lattice::qqmath(pomc_hyp_log_lmer)


anova(pomc_hyp_log_lmer)
emmeans(pomc_hyp_log_lmer, pairwise ~ Condition | Tissue)$contrasts
emmeans(pomc_hyp_log_lmer, pairwise ~ Tissue | Condition)$contrasts

pomc_hyp_log_plot <- ggplot() +
  stat_summary(data=pomc_hyp, aes(x=Condition, y=log(RGE), fill=Condition),
               fun = 'mean', geom="bar", width=0.75) +
  #geom_errorbar(data=pomc_hyp_summ, aes(x=Condition, ymin=mean-sd, ymax=mean+sd), width=0.2) +
  geom_point(data=pomc_hyp, aes(x=Condition, y=log(RGE)), size=2) +
  facet_wrap(~Tissue, scales="free") +
  theme_bw() %+replace%
  theme(legend.position = "none") +
  labs(y="Relative Gene Expression",
       x="")

pomc_hyp_log_plot
```



```{r}
pomc_hyp <- read.csv("../data/06092022-POMC-hypothalamus - Copy.csv",
                     fileEncoding = 'UTF-8-BOM')
pomc_hyp <- pomc_hyp %>% mutate(Condition = as.factor(Condition))
pomc_hyp

pomc_hyp_summ <- pomc_hyp %>% group_by(Condition) %>%
                      summarise(n = n(),
                                mean = mean(RGE),
                                sd = sd(RGE))
pomc_hyp_summ

pomc_hyp_plot <- ggplot() +
  stat_summary(data=pomc_hyp, aes(x=Condition, y=RGE, fill=Condition),
               fun = 'mean', geom="bar", width=0.75) +
  geom_errorbar(data=pomc_hyp_summ, aes(x=Condition, ymin=mean-sd, ymax=mean+sd), width=0.2) +
  geom_point(data=pomc_hyp, aes(x=Condition, y=RGE), size=2) +
  theme_bw() %+replace%
  theme(legend.position = "none") +
  labs(y="Relative Gene Expression",
       x="") +
  scale_y_continuous(expand = expansion(mult = c(0, 0.25)))

pomc_hyp_plot


#bad
pomc_hyp_lm <- lm(RGE ~ Condition, pomc_hyp)
plot(pomc_hyp_lm, which=c(1,2))
#better
pomc_hyp_log_lm <- lm(log(RGE) ~ Condition, pomc_hyp)
plot(pomc_hyp_log_lm, which=c(1,2))

anova(pomc_hyp_log_lm)
t.test(log(RGE) ~ Condition, pomc_hyp)

pomc_hyp_log_plot <- ggplot() +
  stat_summary(data=pomc_hyp, aes(x=Condition, y=log(RGE), fill=Condition),
               fun = 'mean', geom="bar", width=0.75) +
  #geom_errorbar(data=pomc_hyp_summ, aes(x=Condition, ymin=mean-sd, ymax=mean+sd), width=0.2) +
  geom_point(data=pomc_hyp, aes(x=Condition, y=log(RGE)), size=2) +
  theme_bw() %+replace%
  theme(legend.position = "none") +
  labs(y="Relative Gene Expression",
       x="")

pomc_hyp_log_plot
```

