---
title: "qpCR-Figures"
author: "C-T Berezin"
date: "10/30/2021"
output:
  pdf_document: default
  html_document: default
---

```{r, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, error = FALSE)
library(tidyverse)
library(ggplot2)
library(ggthemes)
library(forcats)
library(ggpubr)
library(writexl)
library(svglite)
```

### McKO validation

```{r mcko}
mcko <- read.csv("../data/08162021-McKO-totalret-final.csv", fileEncoding = 'UTF-8-BOM')
mcko <- mcko %>% 
  mutate(Genotype = fct_relevel(Genotype, c("WT", "McKO")))

mcko_summ <- mcko %>% group_by(Genotype) %>%
                      summarise(n = n(),
                                mean = mean(Relative.GE),
                                sd = sd(Relative.GE))

mcko_plot <- ggplot() +
  stat_summary(data=mcko, aes(x=Genotype, y=Relative.GE, fill=Genotype),
               fun = 'mean', geom="bar", width=0.75) +
  geom_errorbar(data=mcko_summ, aes(x=Genotype, ymin=mean-sd, ymax=mean+sd), width=0.2) +
  geom_point(data=mcko, aes(x=Genotype, y=Relative.GE), size=2) +
  theme_bw() %+replace%
  theme(legend.position = "none") +
  labs(y="Relative Gene Expression",
       x="") +
  scale_y_continuous(expand = expansion(mult = c(0, 0.2)))

mcko_plot

#ggsave(filename="../figures/mcko_fig_bars.svg", plot=mcko_plot, height=4, width=3)

mcko %>% group_by(Genotype) %>% 
  summarise(n= n(),
            mean = mean(Relative.GE),
            sd = sd(Relative.GE))
```


### Is POMC under circadian regulation? 

```{r pomc-circadian}
pomc_circ <- read.csv("../data/08202021-POMC-circ-final.csv",
                      fileEncoding = 'UTF-8-BOM')
pomc_circ <- pomc_circ %>% mutate(Time = as.factor(Time),
                                  Sex = as.factor(Sex)) %>% 
  mutate(Time = fct_relevel(Time, c("8am", "4pm", "12am"))) %>%
  mutate(Time = fct_recode(Time,
                           "2" = "8am",
                           "10" = "4pm",
                           "18" = "12am")) %>% 
  filter(Relative.GE < 3) #remove mouse with high relative GE (data not normal)

#p-values from SigmaPlot
pomc_circ_stats <- tibble::tribble(
  ~group1, ~group2, ~group3, ~p.adj,
  "08:00", "16:00", "00:00", "0.865"
)


pomc_circ_plot <- ggplot(pomc_circ, aes(x=Time, y=Relative.GE)) +
  stat_summary(fun = 'mean', geom="bar") +
  stat_summary(fun.data = mean_se, geom = "errorbar", width=.2) +
  geom_point(aes(col=Sex)) +
  theme_few() +
  labs(x="Zeitgeber Time (ZT)",
       y="Relative Gene Expression") +
  geom_bracket(xmin="2", xmax="18", y.position=2.5, label="n.s.") +
  geom_bracket(xmin="10", xmax="18", y.position=2.5, label="") +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1)))
  
pomc_circ_plot

#ggsave(filename="../figures/pomc_circ_fm.png", plot=pomc_circ_plot, height=3, width=3)
```

```{r}
#males only
pomc_circ_males <- pomc_circ %>% filter(Sex == "M")

pomc_circ_m_summ <- pomc_circ_males %>% group_by(Time) %>%
                    summarise(n = n(),
                              mean = mean(Relative.GE),
                              sd = sd(Relative.GE))

pomc_circ_m_summ

pomc_circ_m_plot <- ggplot() +
  stat_summary(data=pomc_circ_males, aes(x=Time, y=Relative.GE), fun = 'mean', geom="bar",
               fill=c("yellow", "yellow", "grey70"),
               width=0.75) +
  geom_errorbar(data=pomc_circ_m_summ, aes(x=Time, ymin=mean-sd, ymax=mean+sd), width=.2) +
  geom_point(data=pomc_circ_males, aes(x=Time, y=Relative.GE), size = 2) +
  theme_bw() %+replace%
  theme(plot.background = element_rect(fill = "transparent",colour = NA),
        panel.background = element_rect(fill = "transparent",colour = NA)) +
  labs(x="Zeitgeber Time (ZT)",
       y="Relative Gene Expression") +
  scale_y_continuous(expand = expansion(mult = c(0, 0.2)))

pomc_circ_m_plot

#ggsave(filename="../figures/pomc_circ_m.svg", plot=pomc_circ_m_plot, height=4, width=4)

#normal
shapiro.test(pomc_circ_males$Relative.GE)

#non-homogenous variance
car::leveneTest(Relative.GE ~ Time, pomc_circ_males)

pomc_circ_m_lm <- lm(Relative.GE ~ Time, pomc_circ_males)

#res vs fitted plots looks OK, non-homogenous variance probably not a big concern
plot(pomc_circ_m_lm, which=c(1,2))

anova(pomc_circ_m_lm)
#no significant comps
emmeans::emmeans(pomc_circ_m_lm, pairwise ~ Time)$contrasts

ctrl_pomc_circ_males <- pomc_circ_males %>% filter(Time == "2")

pomc_circ_m_sumstats <- pomc_circ_males %>%
  group_by(Time) %>%
  summarise(n = n(),
            mean = mean(Relative.GE),
            sd = sd(Relative.GE))

pomc_circ_m_sumstats

#between var
sum(5*(mean(pomc_circ_males$Relative.GE - mean(ctrl_pomc_circ_males$Relative.GE)))^2)
#within.var
anova(pomc_circ_m_lm)["Residuals", "Mean Sq"]
power.anova.test(groups=3, n=5, between.var=0.009116243,
                 within.var=0.1952332, sig.level=0.05)
```

### POMC circadian + light-dark analysis

```{r}
pomc_all <- read.csv("../data/02032022-POMC-circ-ld-gene-study.csv",
                     fileEncoding = 'UTF-8-BOM')
pomc_all <- pomc_all %>% mutate(Sex = as.factor(Sex),
                                Hour = as.factor(Hour),
                                Time = as.factor(Time),
                                Light = as.factor(Light))
pomc_all <- pomc_all %>% mutate(Hour = fct_relevel(Hour, c("8am", "12pm", "4pm", "12am", "1am")),
                                Light = fct_relevel(Light, c("light", "dark")))

qqnorm(pomc_all$RelativeGE)
qqline(pomc_all$RelativeGE)

qqnorm(log2(pomc_all$RelativeGE))
qqline(log2(pomc_all$RelativeGE))
shapiro.test(log2(pomc_all$RelativeGE))

pomc_all_lm <- lm(log2(RelativeGE) ~ Sex * Light * Time, data=pomc_all)
plot(pomc_all_lm, which=c(1))
pomc_all_aov <- anova(pomc_all_lm)
pomc_all_aov
#TukeyHSD(aov(pomc_all_lm))

#but can we actually group all timepoints within day and night
day_pomc_all  <- pomc_all %>% filter(Time == "day")
qqnorm(day_pomc_all$RelativeGE)
qqline(day_pomc_all$RelativeGE)

qqnorm(log2(day_pomc_all$RelativeGE))
qqline(log2(day_pomc_all$RelativeGE))
shapiro.test(log2(day_pomc_all$RelativeGE))
day_pomc_lm <- lm(log2(RelativeGE) ~ Hour, data=day_pomc_all)
anova(day_pomc_lm)

night_pomc_all  <- pomc_all %>% filter(Time == "night")
qqnorm(night_pomc_all$RelativeGE)
qqline(night_pomc_all$RelativeGE)

qqnorm(log2(night_pomc_all$RelativeGE))
qqline(log2(night_pomc_all$RelativeGE))
shapiro.test(log2(night_pomc_all$RelativeGE))
night_pomc_lm <- lm(log2(RelativeGE) ~ Hour, data=night_pomc_all)
anova(night_pomc_lm)
t.test(RelativeGE ~ Hour, data=night_pomc_all)
#indeed

par(mfrow=c(1,2))

pomc_all_mf <- ggplot(pomc_all, aes(x=Time, y=RelativeGE)) +
  theme_few() +
  facet_wrap(~Sex) +
  stat_summary(fun = 'mean', geom="bar") +
  geom_point(aes(col=Light))

pomc_all_mf

#ggsave(filename="../figures/pomc_all_mf.png", plot=pomc_all_mf)
```

```{r}
#males only
m_pomc_all <- pomc_all %>% filter(Sex == "M")

m_pomc_all_summ <- m_pomc_all %>% group_by(Time, Light) %>% 
  summarise(n = n(),
            mean = mean(RelativeGE),
            sd = sd(RelativeGE),
            log_mean = mean(log(RelativeGE)),
            log_sd = sd(log(RelativeGE))
            )

levels(m_pomc_all$Time) <- c("ZTday", "ZTnight")
levels(m_pomc_all_summ$Time) <- c("ZTday", "ZTnight")

pomc_all_m <- ggplot() +
  stat_summary(data=m_pomc_all, aes(x=Light, y=RelativeGE, fill=Light),
               fun = 'mean', geom="bar", width=0.75) +
  geom_point(data= m_pomc_all, aes(x=Light, y=RelativeGE), size=2) +
  geom_errorbar(data=m_pomc_all_summ, aes(x=Light, ymin=mean-sd, ymax=mean+sd), width=.2) +
  facet_wrap(~Time) +
  theme_bw() %+replace%
  theme(legend.position="none",
        axis.text.x = element_text(size = 12),
        strip.text.x = element_text(size=14),
        axis.title.x = element_blank(),
        panel.spacing = unit(0,"pt"),
        plot.background = element_rect(fill = "transparent",colour = NA),
        panel.background = element_rect(fill = "transparent",colour = NA)) +
  scale_fill_manual(values=c("yellow", "grey70")) +
  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +
  labs(y="Relative Gene Expression")

pomc_all_m

#ggsave(filename="../figures/pomc_all_m.svg", plot=pomc_all_m, height=4, width=6)

#non-normal
shapiro.test(m_pomc_all$RelativeGE)
#normal
shapiro.test(log(m_pomc_all$RelativeGE))

car::leveneTest(log(RelativeGE) ~ Light*Time, m_pomc_all)

pomc_all_m_log_lm <- lm(log(RelativeGE) ~ Light*Time, m_pomc_all)

plot(pomc_all_m_log_lm, which=c(1,2))

#no significant factors
anova(pomc_all_m_log_lm)

emmeans::emmeans(pomc_all_m_log_lm, pairwise ~ Light * Time)$contrasts
emmeans::emmeans(pomc_all_m_log_lm, pairwise ~ Light | Time)$contrasts
emmeans::emmeans(pomc_all_m_log_lm, pairwise ~ Time | Light)$contrasts

pomc_all_m_log <- ggplot() +
  stat_summary(data=m_pomc_all, aes(x=Light, y=log(RelativeGE), fill=Light),
               fun = 'mean', geom="bar", width=0.75) +
  geom_point(data=m_pomc_all, aes(x=Light, y=log(RelativeGE)), size=2) +
  geom_errorbar(data=m_pomc_all_summ, aes(x=Light, ymin=log_mean-log_sd, ymax=log_mean+log_sd), width=.2) +
  facet_wrap(~Time) +
  theme_bw() %+replace%
  theme(legend.position="none",
        axis.title.x = element_blank(),
        axis.text.x = element_text(size = 12),
        strip.text.x = element_text(size=14),
        panel.spacing = unit(0,"pt"),
        plot.background = element_rect(fill = "transparent",colour = NA),
        panel.background = element_rect(fill = "transparent",colour = NA)) +
  scale_fill_manual(values=c("yellow", "grey70")) +
  labs(y="log(Relative Gene Expression)") +
  ylim(ymin=-0.75, ymax=2.2)

pomc_all_m_log

#ggsave(filename="../figures/pomc_all_m_log.svg", plot=pomc_all_m_log, height=4, width=6)

ctrl_pomc_all_m <- m_pomc_all %>% filter(Light == "light" & Time == "day")

pomc_all_m_sumstats <- m_pomc_all %>%
  group_by(Light, Time) %>%
  summarise(n = n(),
            mean = mean(RelativeGE),
            log_mean = mean(log(RelativeGE)))

pomc_all_m_sumstats

#between var
sum(3*(mean(m_pomc_all$RelativeGE - mean(ctrl_pomc_all_m$RelativeGE)))^2)
#within.var
anova(pomc_all_m_log_lm)["Residuals", "Mean Sq"]
power.anova.test(groups=4, n=3, between.var=0.0006157598,
                 within.var=0.2920105, sig.level=0.05)
```
